dace/transformation/dataflow/vectorization.py

```diff
@@ -18,100 +143,341 @@ class Vectorization(transformation.SingleStateTransformation):
+                        # Make sure we only have Code->Code copies and from arrays
+                        return False
+
+                    if isinstance(src_node, nodes.AccessNode):
+                        src_desc = src_node.desc(sdfg)
+                        if isinstance(src_desc, data.Stream) and self.target == dtypes.ScheduleType.SVE_Map:
+                            # Stream pops are not implemented
+                            return False
+
+        # Check if it is possible to vectorize data container
+        if self.target == dtypes.ScheduleType.FPGA_Device and self._level == 0:
+            maps_to_vectorize, data_descriptors_to_vectorize = collect_maps_to_vectorize(sdfg, state, map_entry)
+
+            old_map_entry = self.map_entry
+            self._level = 1  # To prevent infinte loop
+
+            for m in maps_to_vectorize:
+                self.map_entry = m
+
+                if not self.can_be_applied(get_innermost_state_for_node(sdfg, m), expr_index,
+                                           get_innermost_sdfg_for_node(sdfg, m), permissive):
+                    return False
+
+            # Check alls strideds of the arrays
+            for a in data_descriptors_to_vectorize:
+
+                correct_sdfg = get_innermost_sdfg_for_array(sdfg, a)
+                array = correct_sdfg.arrays[a]
+                strides_list = list(array.strides)
+
+                if strides_list[-1] != 1:
+                    return False
+
+                strides_list.pop()
+
+                for i in strides_list:
+                    if isInt(i) and i % self.vector_len != 0:
+                        return False
+
+            # Check all maps
+            for m in maps_to_vectorize:
+                real_map: nodes.Map = m.map
+                ranges_list = list(real_map.range)
+
+                if isInt(ranges_list[-1][1]) and (ranges_list[-1][1] + 1) % self.vector_len != 0:
+                    return False
 
-        return found
+                if ranges_list[-1][2] != 1:
+                    return False
+
+            # Check all edges
+            for m in maps_to_vectorize:
+
+                correct_state = get_innermost_state_for_node(sdfg, m)
+
+                edges = correct_state.all_edges(m)
+
+                map_subset = m.map.params
+
+                for e in edges:
+
+                    if e is None or e.data is None or e.data.subset is None:
+                        continue
+
+                    edge_subset = [a_tuple[0] for a_tuple in list(e.data.subset)]
+
+                    if isinstance(edge_subset[-1], symbol) and str(edge_subset[-1]) != map_subset[-1]:
+                        return False
+
+            # Not possible to handle interstate edges
+
+            for e, _ in sdfg.all_edges_recursive():
+
+                if isinstance(e.data, InterstateEdge):
+
+                    if e.data.assignments != {} or e.data.condition.as_string != "1":
+                        return False
```

Comment:
I don't understand this -- does the transformation fail if _any_ interstate edge _anywhere_ in the SDFG has an assignment on it? Or what is happening?